### PR TITLE
Fixes logic used to skip publishes based on predistributor.

### PR DESCRIPTION
**Pull request by dkliban**
Also removes extraneous code from rsync.

re #1887
https://pulp.plan.io/issues/1887
re #1963
https://pulp.plan.io/issues/1963
re #1759
https://pulp.plan.io/issues/1759

### CODE REVIEW

File: server/pulp/server/controllers/repository.py [hunk 1]
```diff
@@ -1025,6 +1027,9 @@ def check_publish(repo_obj, dist_id, dist_inst, transfer_repo, conduit, call_con
     # Skip if a predistributor is configured and the predistributor has not published since the
     # last publish. Skip if content has not changed since last publish and force_full has not been
     # requested and no predistributor is defined.
+    import pydevd
```
Comment (bmbouter) [inline server/pulp/server/controllers/repository.py]: These need to be removed.

**Comment by bmbouter** [comment]
@dkliban I'm doing my hand testing again, but could you ensure that pulp smash passes as much as it does with this commit as with the parent commit?

**Comment by dkliban** [comment]
@bmbouter i will run pulp-smash now

**Comment by bmbouter** [comment]
My hand testing shows this PR resolved the issue we were experiencing. Also the unit tests all pass for me locally. The Jenkins unit test failure looks unrelated and is also failing in other PRs. On that basis I'm adding the LGTM, but wait until pulp smash gives no unexpected errors before merging. @dkliban Thanks!

**Pull request merged**